services:
  proof-server:
    container_name: 'midnight-local-proof-server-0-12-0'
    image: 'midnightnetwork/proof-server:4.0.0'
    logging:
        # driver: none
        driver: local
    ports:
      - "6300:6300"
    environment:
      # RUST_BACKTRACE: "full"
      RUST_LOG: "error"
  indexer:
    container_name: 'midnight-local-chain-indexer-0-12-0'
    image: 'midnightntwrk/indexer-standalone:2.1.1-rc.1-99b0bce4'
    platform: linux/amd64
    ports:
      - '8088:8088'
    command: ['-Dlogback.configurationFile=logback-json-file.xml']
    environment:
      RUST_LOG: "indexer=info,chain_indexer=info,indexer_api=info,wallet_indexer=info,indexer_common=info,fastrace_opentelemetry=off,info"
      APP__INFRA__SECRET: "303132333435363738393031323334353637383930313233343536373839303132"
      APP__INFRA__NODE__URL: "ws://node:9944"
    depends_on:
      node:
        condition: service_started
    volumes:
      - midnight-data-undeployed:/node
  node:
    image: 'midnightnetwork/midnight-node:0.12.0'
    platform: linux/amd64
    container_name: 'midnight-local-node-0-12-0'
    ports:
      - "9944:9944"
    healthcheck:
      test: [ "CMD", "curl", "-f", "http://localhost:9944/health" ]
      interval: 2s
      timeout: 5s
      retries: 5
      start_period: 40s
    environment:
      CFG_PRESET: "dev"
    volumes:
      - midnight-data-undeployed:/node

volumes:
  midnight-data-undeployed: {}
